spec/coffee/directives/api/circle.spec.coffee

```diff
@@ -0,0 +1,129 @@
+describe 'uiGmapCircle', ->
+  allDone =  undefined
+  rootScope = null
+  timeout = null
+  GCircle = null
+  digest = null
+  modelClicked = false
+  browser = null
+
+  afterEach ->
+    digest = null
+    GCircle.resetInstances()
+    modelClicked = false
+
+  beforeEach ->
+    digest = (fn) =>
+     fn()
+     timeout?.flush() if browser.deferredFns?.length
+     rootScope?.$digest()
```

Comment:
Similar to my other comment, if there is some code somewhere (possibly in google's API) that sets a timeout from within a function that is already called from timeout, then it is completely possible you'll have to use a flush/digest loop to let the whole chain get executed.

As another thought, why the `?` guards on timeout and rootScope here?  you've injected them, so you should be able to count on them being defined.  On the other hand, if they're not defined, then you can't expect this code to work correctly with $evalAsync, and the `?` guard will just disguise the fact that `$digest()` isn't actually being called.

This is just me speculating, I would have to take some time to play around with it if I were really going to investigate.